# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/push/PushSubscription.cpp
# Commit: a8c5441fd896
# Full Hash: a8c5441fd89658dea45d939b1e930cf3d148fafd
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-01-26 21:47:24
# Description:
#   Bug 1873573 - Rename PromiseWorkerProxy::WorkerPromise() to GetWorkerPromise(). r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199323
# ==============================================================================

diff -r 533ecc686bac -r a8c5441fd896 dom/push/PushSubscription.cpp
--- a/dom/push/PushSubscription.cpp	Fri Jan 26 10:41:43 2024 +0000
+++ b/dom/push/PushSubscription.cpp	Fri Jan 26 10:48:56 2024 +0000
@@ -70,7 +70,11 @@
     MOZ_ASSERT(aWorkerPrivate);
     aWorkerPrivate->AssertIsOnWorkerThread();
 
-    RefPtr<Promise> promise = mProxy->WorkerPromise();
+    RefPtr<Promise> promise = mProxy->GetWorkerPromise();
+    // Once Worker had already started shutdown, workerPromise would be nullptr
+    if (!promise) {
+      return true;
+    }
     if (NS_SUCCEEDED(mStatus)) {
       promise->MaybeResolve(mSuccess);
     } else {